### PR TITLE
DROOLS-3308 DMN implement missing functions only described in ...

### PR DESCRIPTION
**Pull request by tarilabs**
...chapter "10.3.2.6 Context"

@etirelli wdyt?

Also, once we fully confirm this, I would then raise an issue on the OMG JIRA: in order to reach consensus for instance on TCK for calling these 2 built-in functions by binding parameters by name, I believe is important to agree on the name of such parameters by describing these functions in section "10.3.4 Built-in functions" by making a specific sub-section for Context. Wdyt ?

/cc @baldimir @jiripetrlik

### CODE REVIEW

File: kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java [hunk 1]
```diff
@@ -2402,5 +2402,23 @@ public void testAnyExpression() {
         assertThat(dmnResult.getDecisionResultByName("Decision-1").getEvaluationStatus(), is(DecisionEvaluationStatus.SUCCEEDED));
         assertThat(dmnResult.getDecisionResultByName("Decision-1").getResult(), is("Hello World"));
     }
+
+    @Test
+    public void testGetEntriesGetValue() {
```
Comment (baldimir) [inline kie-dmn/kie-dmn-core/src/test/java/org/kie/dmn/core/DMNRuntimeTest.java]: Could you please separate this into two tests? One for getEntries and one for getValue. In that case if both things fail, we will see both failures. Currently fail in getEntries assert hides potential getValue issue.

File: kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/FEELFunctionsTest.java [hunk 2]
```diff
@@ -150,7 +150,10 @@
                                                                                  BigDecimal.valueOf( 4 ), BigDecimal.valueOf( 5 ) ), null },
                 { "sort( [\"c\", \"e\", \"d\", \"a\", \"b\"], function(x,y) x < y )", Arrays.asList( "a", "b", "c", "d", "e" ) , null},
                 { "sort( list : [\"c\", \"e\", \"d\", \"a\", \"b\"], precedes : function(x,y) x < y )", Arrays.asList( "a", "b", "c", "d", "e" ) , null},
-                { "sort( precedes : function(x,y) x < y, list : [\"c\", \"e\", \"d\", \"a\", \"b\"] )", Arrays.asList( "a", "b", "c", "d", "e" ) , null}
+                { "sort( precedes : function(x,y) x < y, list : [\"c\", \"e\", \"d\", \"a\", \"b\"] )", Arrays.asList( "a", "b", "c", "d", "e" ) , null},
```
Comment (baldimir) [inline kie-dmn/kie-dmn-feel/src/test/java/org/kie/dmn/feel/runtime/FEELFunctionsTest.java:153]: Please add also a negative test case(s). E.g. trying to get what is not in the context for both getEntries and getValue.

**Comment by etirelli** [comment]
I agree, we need to open a ticket to the RTF to include these functions in the function list explicitly.

**Review by baldimir — APPROVED** [review]

**Comment by tarilabs** [comment]
```
org.drools.compiler.integrationtests.CompositeAgendaTest.testCreateHaltDisposeAgenda

Failing for the past 1 build (Since Unstable#275 )
Took 34 sec.
Error Message
test timed out after 20000 milliseconds
Stacktrace
org.junit.runners.model.TestTimedOutException: test timed out after 20000 milliseconds
	at java.net.URLStreamHandler.parseURL(URLStreamHandler.java:254)
	at sun.net.www.protocol.file.Handler.parseURL(Handler.java:67)
	at java.net.URL.<init>(URL.java:622)
	at java.net.URL.<init>(URL.java:490)
	at sun.misc.URLClassPath$FileLoader.getResource(URLClassPath.java:1260)
	at sun.misc.URLClassPath.getResource(URLClassPath.java:239)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:365)
	at java.net.URLClassLoader$1.run(URLClassLoader.java:362)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.net.URLClassLoader.findClass(URLClassLoader.java:361)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:424)
	at sun.misc.Launcher$AppClassLoader.loadClass(Launcher.java:349)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:357)
	at org.drools.core.common.CompositeDefaultAgenda.halt(CompositeDefaultAgenda.java:233)
```

Jenkins test again

**Pull request merged**